src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.cs

```diff
@@ -32,6 +32,13 @@ protected override async Task<InsertionPoint> GetInsertionPointAsync(SemanticDoc
             var root = await document.Document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
             var basePosition = root.FindToken(position);
 
+            // Check if we are extracting a local function and are within a local function
+            var localMethodNode = basePosition.GetAncestor<LocalFunctionStatementSyntax>();
```

Comment:
nit: It's a minor improvement to check the boolean before walking up the syntax tree. 

```
if (ExtractLocalFunction) 
{
    var localMethodNode = basePosition.GetAncestor<LocalFunctionStatementSyntax>();
    ...
}
```